src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -889,6 +889,48 @@
     end
   end
 
+  describe 'POST #save_prjconf' do
+    before do
+      login user
+    end
+
+    context 'can save a project config' do
+      before do
+        post :save_prjconf, params: { project: user.home_project.name, submit: 'config', config: 'save config' }
+      end
+
+      it { expect(flash[:success]).to eq('Config successfully saved!') }
+      it { expect(response.status).to eq(200) }
+    end
+
+    context 'cannot save a project config' do
+      before do
+        allow_any_instance_of(ProjectConfigFile).to receive(:save).and_return(nil)
+        post :save_prjconf, params: { project: user.home_project.name, submit: 'config', config: '' }
+      end
+
+      it { expect(flash[:error]).not_to eq(nil) }
+      it { expect(response.status).to eq(400) }
+    end
+
+    context 'cannot save with an unauthorized user' do
+      before do
+        post :save_prjconf, params: { project: another_project.name, submit: 'config', config: 'save config' }
+      end
+
+      it { expect(flash[:error]).to eq('Sorry, you are not authorized to update this Project.') }
+      it { expect(response.status).to eq(302) }
+      it { expect(response).to redirect_to(root_path) }
+    end
+
+    context 'with an non existing project' do
+      it 'raise a RecordNotFound Exception' do
+        params = { project: 'non:existing:project', submit: 'config', config: 'save config' }
+        expect{post :save_prjconf, params: params}.to raise_error ActiveRecord::RecordNotFound
```

Comment:
I would prefer the post to be in a before section. It is more readable in my opinion as it is like that everywhere.